ansible/roles_ocp_workloads/ocp4_workload_bookbag_user/tasks/workload.yaml

```diff
@@ -0,0 +1,72 @@
+---
+
+- name: Get User Data
+  set_fact:
+    _user_data: >-
+      {{ lookup('file', hostvars.localhost.output_dir ~ '/user-data.yaml', errors='ignore')
+      | from_yaml }}
+
+- name: Fallback for User Data
+  when: _user_data | default({}) | length == 0
+  set_fact:
+    _user_data: []
+
+- name: Get User Info
+  set_fact:
+    _user_info_messages: >-
+      {{ lookup('file', hostvars.localhost.output_dir ~ '/user-info.yaml', errors='ignore')
+      | from_yaml }}
+
+- name: Fallback for User Info
+  when: _user_info_messages | default({}) | length == 0
+  set_fact:
+    _user_info_messages: []
+
+- name: Set workshop default vars
+  set_fact:
+    _workshop_default_vars:
+      guid: "{{ guid }}"
+      username: "{{ ocp_username }}"
+      user_info_messages: "{{ _user_info_messages | join('\n') }}"
+
+- name: Set Workshop vars
+  set_fact:
+    _workshop_vars: >-
+      {{ _user_data | combine(_workshop_default_vars) }}
+
+- name: Create Bookbag resources
+  k8s:
+    state: present
+    definition: "{{ lookup('template', item ) | from_yaml }}"
+  loop:
+  - ./templates/project.yaml.j2
+  - ./templates/service_account.yaml.j2
+  - ./templates/rolebinding.yaml.j2
+  - ./templates/service.yaml.j2
+  - ./templates/route.yaml.j2
+  - ./templates/deployment.yaml.j2
```

Comment:
I've been thinking about this as I'm putting together a demo...

I don't think I like using Jinja here. The trouble is that folks doing bookbag development won't be using Jinja, they will be using the OpenShift template in the demo repo. If our deployment doesn't use the same approach then it is bound to diverge and cause problems down the road.

A particular concern is use of the OpenShift console in bookbag. The deployment of the console involves using an init-container and another regular container in the bookbag pod. So that will make for pretty big differences between how some labs are deployed, and more potential trouble if we are not letting the bookbag developer control the deployment definition.

Reply:
I disagree. We should not give folks a template to customize. They can customize the content via the workshop directory. And they can customize the image via the Dockerfile. What else could they need?

And init containers?? These YAML templates are 1:1 copies from the example repo template... There most definitely is no init container: https://github.com/redhat-gpte-devopsautomation/bookbag-template/blob/master/deploy-template.yaml#L65

Reply:
Added init Container